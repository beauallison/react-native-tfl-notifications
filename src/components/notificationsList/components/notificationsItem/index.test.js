import React from 'react';
import { shallow } from 'enzyme';
import NotificationsItem from './index';

describe('components/notificationsList/notificationsItem', () => {
  const props = {
    station: 'London',
    updateStation: () => { },
  };

  it('should render notifications checkbox', () =>
    expect(shallow(<NotificationsItem {...props} />))
      .toMatchSnapshot());

  it('should toggle checkbox', async () => {
    const wrapper = shallow(<NotificationsItem {...props} />);
    wrapper.find('CheckBox').at(0).simulate('press');

    setImmediate(() => expect(wrapper.state('enabled')).toBeTruthy());
  });
});
