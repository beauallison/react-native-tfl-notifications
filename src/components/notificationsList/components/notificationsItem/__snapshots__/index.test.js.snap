// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/notificationsList/notificationsItem should render notifications checkbox 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationsItem
    enabled={false}
    station="London"
    updateStation={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <CheckBox
        center={false}
        checked={false}
        checkedColor="green"
        checkedIcon="check-square-o"
        iconRight={false}
        onPress={[Function]}
        right={false}
        size={24}
        title="London"
        uncheckedColor="#bfbfbf"
        uncheckedIcon="square-o"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "center": false,
        "checked": false,
        "checkedColor": "green",
        "checkedIcon": "check-square-o",
        "iconRight": false,
        "onPress": [Function],
        "right": false,
        "size": 24,
        "title": "London",
        "uncheckedColor": "#bfbfbf",
        "uncheckedIcon": "square-o",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <CheckBox
          center={false}
          checked={false}
          checkedColor="green"
          checkedIcon="check-square-o"
          iconRight={false}
          onPress={[Function]}
          right={false}
          size={24}
          title="London"
          uncheckedColor="#bfbfbf"
          uncheckedIcon="square-o"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "center": false,
          "checked": false,
          "checkedColor": "green",
          "checkedIcon": "check-square-o",
          "iconRight": false,
          "onPress": [Function],
          "right": false,
          "size": 24,
          "title": "London",
          "uncheckedColor": "#bfbfbf",
          "uncheckedIcon": "square-o",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
